stateproof/verify/stateproof.go

```diff
@@ -136,50 +137,54 @@ func GetProvenWeight(votersHdr bookkeeping.BlockHeader, latestRoundInProofHdr bo
 	return provenWeight, nil
 }
 
-// ValidateStateProof checks that a state proof is valid.
-func ValidateStateProof(latestRoundInIntervalHdr *bookkeeping.BlockHeader, stateProof *stateproof.StateProof, votersHdr *bookkeeping.BlockHeader, nextStateProofRnd basics.Round, atRound basics.Round, msg *stateproofmsg.Message) error {
-	proto := config.Consensus[latestRoundInIntervalHdr.CurrentProtocol]
-
-	if proto.StateProofInterval == 0 {
-		return fmt.Errorf("rounds = %d: %w", proto.StateProofInterval, errStateProofNotEnabled)
-	}
-
-	if latestRoundInIntervalHdr.Round%basics.Round(proto.StateProofInterval) != 0 {
-		return fmt.Errorf("state proof at %d for non-multiple of %d: %w", latestRoundInIntervalHdr.Round, proto.StateProofInterval, errNotAtRightMultiple)
+// ValidateStateProof validates that the cryptographic proof of the state proof transaction is correct with respect to the headers.
+func ValidateStateProof(txgroup []transactions.SignedTxn, fetcher BlockHeaderFetcher) error {
+	for i := 0; i < len(txgroup); i++ {
```

Comment:
How can the stateproof transaction be part of a group with more than 1 transaction?